### PR TITLE
chore: Remove last processed height

### PR DESCRIPTION
**Pull request by gitferry**
The `lastProcessedHeight` was introduced for fast catch up blocks for which the fp does not have voting power. However, this case rarely happens and is unharmful. More discussions can be found [here](https://github.com/babylonlabs-io/finality-provider/pull/120#discussion_r1836699698).

### CODE REVIEW

File: finality-provider/service/fp_instance.go [hunk 1]
```diff
@@ -685,9 +675,8 @@ func (fp *FinalityProviderInstance) getPollerStartingHeight() (uint64, error) {
 		startHeight = fppath.MaxUint64(startHeight, latestFinalisedBlocks[0].Height+1)
 	}
 
-	// consider the height after the last processed block
-	lastProcessedHeight := fp.GetLastProcessedHeight()
-	startHeight = fppath.MaxUint64(startHeight, lastProcessedHeight+1)
+	// consider the height after the last voted height
+	startHeight = fppath.MaxUint64(startHeight, fp.GetLastVotedHeight()+1)
```
Comment (Lazar955) [inline finality-provider/service/fp_instance.go:679]:
```suggestion
	startHeight = max[uint64](startHeight, fp.GetLastVotedHeight()+1)
```

Comment (Lazar955) [reply finality-provider/service/fp_instance.go:679]: Nit, let's remove this util function and use go's built-in one. I see that we always pass 2 arguments anyways

**Review by Lazar955 — APPROVED** [review]
Nice cleanup!

File: finality-provider/service/fp_instance.go (hunk 1 above)
Comment (gitferry) [reply finality-provider/service/fp_instance.go]: Good point!

**Pull request merged**